From 5db365ad8aa7569c1c1c8a485e2aef35f152fd95 Mon Sep 17 00:00:00 2001
From: althafvly <althafvly@gmail.com>
Date: Wed, 16 Apr 2025 16:46:11 +0000
Subject: Migrate to murena search from spot

Needed for apps where browser search engine is used for search in apps.
For example BlissLauncher3

Signed-off-by: Saalim Quadri <danascape@gmail.com>
---
 .../chrome/browser/provider/SearchEngineProvider.java    | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/provider/SearchEngineProvider.java b/chrome/android/java/src/org/chromium/chrome/browser/provider/SearchEngineProvider.java
index 299562413fe05..2d1741e8f6b3d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/provider/SearchEngineProvider.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/provider/SearchEngineProvider.java
@@ -62,7 +62,14 @@ public class SearchEngineProvider extends ContentProvider {
             SharedPreferences sharedPreferences = getContext().getSharedPreferences(
                     CONFIGURATION_PREFERENCE_FILE_NAME, Context.MODE_PRIVATE);
             if (sharedPreferences != null) {
-                rowBuilder.add(sharedPreferences.getString(PREFERENCE_KEY, ""));
+                String searchEngine = sharedPreferences.getString(PREFERENCE_KEY, "");
+                if ("espot".equals(searchEngine)) {
+                    // Update to murena search in preferences if it was espot
+                    sharedPreferences.edit().putString(PREFERENCE_KEY, "murenasearch").apply();
+                    rowBuilder.add("murenasearch");
+                } else {
+                    rowBuilder.add(searchEngine);
+                }
             } else {
                 rowBuilder.add("");
             }
-- 
2.55.0

